### PR TITLE
adjust file permissions on windows

### PR DESCRIPTION
**Pull request by ashirley**
This is done by the standard docker client so that files which need to be
executable are even though that isn't suported on windows.

https://github.com/moby/moby/blob/94b8a116fbf1cd90e68d8f5361b520d326a66f9b/pkg/archive/archive_windows.go#L43

Change-Type: patch

### CODE REVIEW

File: lib/utils/docker.coffee [hunk 1]
```diff
@@ -418,3 +428,13 @@ copyQemu = (context) ->
...
+
+# if required, change the permissions (mode) of files added to the tar stream
+# This is needed for windows as the execute bit isn't supported there
+chmodTarEntry = (mode) ->
+	os = require('os')
+	if os.platform() == 'win32'
+		mode |= 0o111
+		mode &= 0o755
+	else
+		mode
```
Comment (ashirley) [inline lib/utils/docker.coffee]: I'm not too familiar with coffee script and this was the only way I could find to return `mode` unchanged on non-windows platforms. If there is a more idiomatic way of doing it, I'll change it.

File: lib/utils/docker.coffee [hunk 2]
```diff
@@ -418,3 +428,13 @@ copyQemu = (context) ->
 	.then ->
 		fs.chmod(binPath, '755')
 	.return(binPath)
+
+# if required, change the permissions (mode) of files added to the tar stream
+# This is needed for windows as the execute bit isn't supported there
+chmodTarEntry = (mode) ->
```
Comment (pimterry) [inline lib/utils/docker.coffee]: Can you add a comment linking to the corresponding moby code for this that you've included in the PR description?

File: lib/utils/docker.coffee [hunk 3]
```diff
@@ -196,13 +196,23 @@ exports.runBuild = (params, options, getBundleInfo, logger) ->
...
 	qemuPath = ''
 
+	if os.platform() == 'win32'
+		logger.logWarn '''
+			SECURITY WARNING: You are building a Docker image from
+			Windows against a non-Windows Docker host.
+			All files and directories added to build context will have
+			'-rwxr-xr-x' permissions. It is recommended to double check
+			and reset permissions for sensitive files and directories.
+		'''
```
Comment (pimterry) [inline lib/utils/docker.coffee]: Can you add a comment here explaining that this is imitating the Docker client behaviour too, and link to the corresponding code for that?

File: lib/utils/docker.coffee (hunk 1 above)
Comment (pimterry) [reply lib/utils/docker.coffee]: There's a variety of other ways to do this, but this is fine imo, don't worry about it (we're in the process of moving this to TypeScript anyway, so the exact details of this kind of thing don't matter too much)

**Comment by pdcastro** [comment]
I understand that this PR addresses issue #749 (emulated builds on Windows). Without realising that this PR existed, I fixed the issue in another PR, #1464. Given that this PR appears stale (almost 2 years old), I am taking the liberty of closing it.

**Pull request closed without merging**